src/parser/expressions_parser.rs

```diff
@@ -305,10 +317,12 @@ fn parse_array_literal(lexer: &mut ParseSession) -> Result<AstNode, Diagnostic>
     let start = lexer.range().start;
     lexer.expect(KeywordSquareParensOpen)?;
     lexer.advance();
+    lexer.paren = true;
     let elements = Some(Box::new(parse_expression(lexer)));
     let end = lexer.range().end;
     lexer.expect(KeywordSquareParensClose)?;
```

Comment:
if this result-unwrap escapes the method, the paren remains true.  will this become a problem if the parser recoveres (e.g. somwhere in a parse_in_region)?

Reply:
Good catch, didn't think about this at all tbh. There also another sideffect happening with nested arrays, e.g. `[(..., [(...)])]` where the inner array will set the flag to false and in turn no ParenExpr is generated for the outer expression. This is easily fixable but Ghaith and I talked about ParenExpr in general and decided to introduce it globally. Hence this PR is back in draft-mode. Thanks for the review though!